File: myoro_matchup_api/src/main/java/com/myoro/myoro_matchup_api/dto/FriendRequestResponseDto.java
package com.myoro.myoro_matchup_api.dto;

import com.myoro.myoro_matchup_api.enums.FriendRequestStatusEnum;

import java.time.LocalDateTime;

/** Friend request response DTO. */
public class FriendRequestResponseDto {
  /** Friend request ID. */
  private Long id;

  /** Name of the user who sent the friend request. */
  private String requesterName;

  /** Name of the user receiving the friend request. */
  private String recipientName;

  /** Status of the friend request. */
  private FriendRequestStatusEnum status;

  /** When the friend request was created. */
  private LocalDateTime createdAt;

  /** When the recipient responded. */
  private LocalDateTime respondedAt;

  /** Default constructor. */
  public FriendRequestResponseDto() {
  }

  /**
   * Getter for id
   * 
   * @return the friend request ID
   */
  public Long getId() {
    return id;
  }

  /**
   * Getter for requesterName
   * 
   * @return the name of the user who sent the friend request
   */
  public String getRequesterName() {
    return requesterName;
  }

  /**
   * Getter for recipientName
   * 
   * @return the name of the user receiving the friend request
   */
  public String getRecipientName() {
    return recipientName;
  }

  /**
   * Getter for status
   * 
   * @return the status of the friend request
   */
  public FriendRequestStatusEnum getStatus() {
    return status;
  }

  /**
   * Getter for createdAt
   * 
   * @return when the friend request was created
   */
  public LocalDateTime getCreatedAt() {
    return createdAt;
  }

  /**
   * Getter for respondedAt
   * 
   * @return when the recipient responded
   */
  public LocalDateTime getRespondedAt() {
    return respondedAt;
  }

  /**
   * Setter for id
   * 
   * @param id the friend request ID
   */
  public void setId(Long id) {
    this.id = id;
  }

  /**
   * Setter for requesterName
   * 
   * @param requesterName the name of the user who sent the friend request
   */
  public void setRequesterName(String requesterName) {
    this.requesterName = requesterName;
  }

  /**
   * Setter for recipientName
   * 
   * @param recipientName the name of the user receiving the friend request
   */
  public void setRecipientName(String recipientName) {
    this.recipientName = recipientName;
  }

  /**
   * Setter for status
   * 
   * @param status the status of the friend request
   */
  public void setStatus(FriendRequestStatusEnum status) {
    this.status = status;
  }

  /**
   * Setter for createdAt
   * 
   * @param createdAt when the friend request was created
   */
  public void setCreatedAt(LocalDateTime createdAt) {
    this.createdAt = createdAt;
  }

  /**
   * Setter for respondedAt
   * 
   * @param respondedAt when the recipient responded
   */
  public void setRespondedAt(LocalDateTime respondedAt) {
    this.respondedAt = respondedAt;
  }
}

File: myoro_matchup/lib/module/friend/dto/friend_request_response_dto.dart
import 'package:faker/faker.dart';
import 'package:flutter/foundation.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:myoro_flutter_annotations/myoro_flutter_annotations.dart';
import 'package:myoro_flutter_library/myoro_flutter_library.dart';
import 'package:myoro_matchup/myoro_matchup.dart';

part 'friend_request_response_dto.g.dart';

/// Friend request response DTO.
@immutable
@myoroModel
@JsonSerializable(createToJson: false)
final class FriendRequestResponseDto with _$FriendRequestResponseDtoMixin {
  /// Default constructor.
  const FriendRequestResponseDto({
    required this.id,
    required this.requesterName,
    required this.recipientName,
    required this.status,
    required this.createdAt,
    required this.respondedAt,
  });

  /// From JSON.
  factory FriendRequestResponseDto.fromJson(Map<String, dynamic> json) {
    return _$FriendRequestResponseDtoFromJson(json);
  }

  /// Fake constructor.
  factory FriendRequestResponseDto.fake() {
    final createdAt = myoroFake<DateTime>();
    final respondedAt = createdAt.add(const Duration(days: 1));

    return FriendRequestResponseDto(
      id: faker.randomGenerator.integer(100000, min: 1),
      requesterName: faker.person.name(),
      recipientName: faker.person.name(),
      status: FriendRequestStatusEnum.fake(),
      createdAt: createdAt,
      respondedAt: respondedAt,
    );
  }

  /// Friend request identifier.
  final int id;

  /// Name of the user who sent the friend request.
  final String requesterName;

  /// Name of the user receiving the friend request.
  final String recipientName;

  /// Status of the friend request.
  final FriendRequestStatusEnum status;

  /// When the friend request was created.
  final DateTime createdAt;

  /// When the recipient responded.
  final DateTime? respondedAt;
}
